pyblish/plugin.py

```diff
@@ -780,6 +782,46 @@ def current_host():
     return pyblish._registered_hosts[-1] or "unknown"
 
 
+def register_callback(signal, callback):
+
+    if signal in pyblish._registered_callbacks:
+        pyblish._registered_callbacks[signal].append(callback)
+    else:
+        pyblish._registered_callbacks[signal] = [callback]
+
+
+def deregister_callback(callback):
+
+    for signal in pyblish._registered_callbacks:
+        if callback in pyblish._registered_callbacks[signal]:
+            pyblish._registered_callbacks[signal].remove(callback)
+
+
+def deregister_callbacks(signal):
+
+    pyblish._registered_callbacks.pop(signal, None)
+
+
+def deregister_all_callbacks():
+
+    pyblish._registered_callbacks = dict()
+
+
+def registered_callbacks():
+
+    return pyblish._registered_callbacks
+
+
+def emit(signal, **kwargs):
+
+    if signal in pyblish._registered_callbacks:
```

Comment:
How about this?

```python
for callback in pyblish._registered_callbacks.get(signal, []):
  try:
    callback(**kwargs)
 except Exception as e:
   log.error(e)
```

That way, nothing happens if there aren't any of those signals, and we reduce one line and one level of indentation.

Also, what *should* happen when a callback fails? I would not silence it. I would let it roar.

```python
for callback in pyblish._registered_callbacks.get(signal, []):
  try:
    callback(**kwargs)
 except Exception as e:
   # Print full exception message, but don't break the loop
   traceback.print_exc(e)
```

This way you still get information on which line and file the error occurred, as opposed to just the final message.